platform/darwin/include/MGLStyle.h

```diff
@@ -4,50 +4,115 @@
 
 NS_ASSUME_NONNULL_BEGIN
 
-/** A collection of convenience methods for creating style URLs of default styles provided by Mapbox. These instances of NSURL are cached. */
+/**
+ A version number identifying the latest released version of the suite of default styles provided by Mapbox. This version number may be passed into one of the “StyleURLWithVersion” class methods of MGLStyle.
+ 
+ The value of this constant is current as of the date on which this SDK was published. Consult the <a href="https://www.mapbox.com/api-documentation/#styles">Mapbox Styles API documentation</a> for the most up-to-date style versioning information.
+ 
+ @warning The value of this constant may change in a future release of the SDK. If you use any feature that depends on a specific implementation detail in a default style, you may use the current value of this constant or the underlying style URL, but do not use the constant itself. Such implementation details may change significantly from version to version.
```

Comment:
Let me go on the record stating that I think including a `MGLStyleCurrentVersion` constant is a trap and that we should require people to explicitly specify a version number.

But it we do include it, we should not use the term "implementation detail" -- this suggests that we don't consider style features to be part of an API contract, when in fact the very reason we are using versions is to indicate changes to documented semantics.